packaging/dependency/mongo.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dependency
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/packaging"
+)
+
+type mongoDependency struct {
+	useNUMA bool
+}
+
+// Mongo returns a dependency for installing mongo server using the specified
+// NUMA settings.
+func Mongo(useNUMA bool) packaging.Dependency {
+	return mongoDependency{useNUMA: useNUMA}
+}
+
+// PackageList implements packaging.Dependency.
+func (dep mongoDependency) PackageList(series string) ([]packaging.Package, error) {
+	var pkgList []string
+	var pm = packaging.AptPackageManager
+
+	if dep.useNUMA {
+		pkgList = append(pkgList, "numactl")
+	}
+
+	switch series {
+	case "centos7", "opensuseleap":
+		return nil, errors.NotImplementedf("installing mongo on %s", series)
+	case "precise":
```

Comment:
similarly drop 'precise' and 'zesty' 'artful', I don't know if we need an explicit 'disco' but I don't think we do.

Reply:
I don't think we need a 'disco' one we should hit the default block for everything else. If at some point we decide to use snaps for mongo we can add a case for bionic to use apt and make the default block use snap.

Reply:
I would follow the same pattern as above so it is predictable. If we decided to do explicit disco/eon then we'd want to do it in both places.